File: LSP/Kata_ Shop Omnichannel/SolucionPhp/Shop/Persistence/ShopSQLPersistence.php
<?php

declare(strict_types=1);

namespace SolucionPhp\Shop\Persistence;

use SolucionPhp\Shop\MiniShop;

class ShopSQLPersistence
{
    private MiniShop $shop;
    //private SQLPersistence $persistence;

    public function __construct(MiniShop $shop)
    {
        $this->shop = $shop;
        //$this->persistence = new SQLPersistence();
    }

    public function save(): void
    {
        //$stmt = $this->persistence->prepare("INSERT INTO shops VALUES (:name, :phone)");
        /*$stmt->execute([
            'name' => $this->shop->getName(),
            'phone' => $this->shop->getPhone()
        ]);*/
    }
}

File: SRP/Kata_ Shop/SolucionJava/Shop/Persistence/ShopSQLPersistence.java
package SolucionJava.Shop.Persistence;

public class ShopSQLPersistence {
    private Shop shop;
    //private SQLPersistence persistence;

    public ShopSQLPersistence(Shop shop){
        this.shop = shop;
        //this.persistence = new SQLPersistence();
    }

    public void save(){
        // Statement stmt = this.persistence.prepareQuery("INSERT INTO shops VALUES (?, ?)");
        // stmt.setString(1, this.shop.getName());
        // stmt.setInteger(2, this.shop.getPhone());
        // stmt.execute();
    }
}
